cmake/HalideGeneratorHelpers.cmake

```diff
@@ -100,11 +125,10 @@ function(add_halide_generator TARGET)
                "#include \"Halide.h\"\n"
                "HALIDE_GENERATOR_PYSTUB(${GEN_NAME}, ${MODULE_NAME})\n")
 
-        file(WRITE
-             "${CMAKE_CURRENT_BINARY_DIR}/${TARGET}.py_stub_generated.cpp"
-             "${stub_text}")
+        set(generated_cpp "${CMAKE_CURRENT_BINARY_DIR}/${TARGET}.py_stub_generated.cpp")
+        file(WRITE "${generated_cpp}" "${stub_text}")
 
-        Python3_add_library(${TARGET}_pystub MODULE WITH_SOABI "${CMAKE_CURRENT_BINARY_DIR}/${TARGET}.py_stub_generated.cpp" ${ARG_SOURCES})
+        Python3_add_library(${TARGET}_pystub MODULE WITH_SOABI "${generated_cpp}" ${ARG_SOURCES})
```

Comment:
Hm, if building WITH_PYTHON_BINDINGS=OFF then yes, I suppose. But this PR builds just fine that way (several of the buildbots only test that way), probably because we never attempt to use any of the python-related stuff when this is disabled. Not sure if we should wrap this (and other) sections in WITH_PYTHON_BINDINGS checks, vs just let people's code fail to build when they try to use Python with a non-Python Halide.

Reply:
Actually, yeah, it's definitely possible for out-of-tree builds (eg apps): if they don't explicitly call it, it won't have been called. So the question is: should we do so here lazily? Or (alternately) do something like

`if (NOT TARGET Halide::Python) message(FATAL_ERROR "You must call find_package(Python3) in order to use this rule with Python Generators"`

Reply:
Resolved (I think) by b789450926c2ca64bad5700bed4faae11be8baa3